### ISSUE TITLE
`vue/no-mutating-props`: extend to detect nested mutation of `defineModel()` refs

### ISSUE THREAD

**Issue by faizanu94** [vuejs/eslint-plugin-vue#3130]
**What rule do you want to change?**

`vue/no-mutating-props`

**Does this change cause the rule to produce more or fewer warnings?**

More. It adds detection for a case that currently passes silently.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default behavior (no new option). `defineModel()` is sugar for a `modelValue` prop + `update:modelValue` emit, so mutating it in place is the same category of bug the rule already catches for `defineProps()`. This mirrors the precedent set by #2360 and #3032, which taught other rules about `defineModel()`.

**Please provide some example code that this change will affect:**

```vue
<script setup>
const model = defineModel()

function bad() {
  model.value.name = 'x'       // nested property mutation
  model.value.items[0] = 'y'   // nested index assignment
  model.value.items.push('z')  // mutating array method
}

function good() {
  model.value = { ...model.value, name: 'x' } // correct: reassigns the ref
}
</script>
```

**What does the rule currently do for this code?**

Nothing — 0 errors. Verified by running the current rule (v10.10.0) via ESLint's `Linter` against the parsed SFC. For comparison, the *identical* mutation shape on a `defineProps()` object IS flagged:

| Case | Current result |
| --- | --- |
| `defineModel()` nested mutation (the 3 `bad()` lines) | 0 errors |
| `defineProps(['user'])` then `props.user.name = 'x'` / `.push()` | 2 errors |
| `defineModel()` correct pattern (`model.value = {...}`) | 0 errors |

Root cause: the rule only registers `onDefinePropsEnter`; there is no `onDefineModelEnter` handler, even though that visitor hook already exists.

**What will the rule do after it's changed?**

Flag the three `bad()` lines as `unexpectedMutation`, while leaving `good()` (and the `shallowOnly` option semantics) untouched.

**Additional context**

Why it's a real bug, not style: `model.value = next` goes through the ref setter and emits `update:modelValue`, syncing the parent. `model.value.foo = x` skips the setter, so no emit fires and the parent silently desyncs.

Scope note: this is a syntactic check on direct `model.value.<path>` mutation. It will not track mutation via an intermediate alias (`const i = model.value.items[0]; i.x = 1`) — that needs data-flow analysis and I'd suggest keeping it out of scope for v1.

I'm happy to open the PR (implementation + tests + docs + changeset) if this is accepted.

**Comment by faizanu94** [comment]
Opened a draft PR implementing this: #3131.

It wires `defineModel()` bindings into the existing `findMutating` pipeline via `onDefineModelEnter` (same pattern as #2360 / #3032). Reassigning the ref (`model.value = next`) stays allowed; only nested mutation is reported, and `shallowOnly` still permits it. Includes tests, docs, and a changeset — full suite green.

Left it as a draft pending your call on the default-on behavior. Happy to adjust.
